src/util/Util.cpp

```diff
@@ -37,34 +37,37 @@ static bool execInUiThreadCallback(CallbackUiData* cb)
  *
  * Make sure the container class is not deleted before the UI stuff is finished!
  */
-void Util::execInUiThread(std::function<void()> callback)
+void Util::execInUiThread(std::function<void()>&& callback)
 {
-	gdk_threads_add_idle((GSourceFunc) execInUiThreadCallback, new CallbackUiData(callback));
+	gdk_threads_add_idle((GSourceFunc) execInUiThreadCallback, new CallbackUiData(std::move(callback)));
 }
 
-void Util::cairo_set_source_rgbi(cairo_t* cr, int c)
+void Util::cairo_set_source_rgbi(cairo_t* cr, int color)
 {
-	double r = ((c >> 16) & 0xff) / 255.0;
-	double g = ((c >> 8) & 0xff) / 255.0;
-	double b = (c & 0xff) / 255.0;
+	auto tmp = static_cast<unsigned>(color);
+
+	double r = ((tmp >> 16u) & 0xffu) / 255.0;
+	double g = ((tmp >> 8u) & 0xffu) / 255.0;
+	double b = (tmp & 0xffu) / 255.0;
 
 	cairo_set_source_rgb(cr, r, g, b);
 }
 
 
 void Util::apply_rgb_togdkrgba(GdkRGBA& col, int color)
 {
-	col.red = ((color >> 16) & 0xFF) / 255.0;
-	col.green = ((color >> 8) & 0xFF) / 255.0;
-	col.blue = (color & 0xFF) / 255.0;
+	auto tmp = static_cast<unsigned>(color);
+	col.red = ((tmp >> 16u) & 0xFFu) / 255.0;
+	col.green = ((tmp >> 8u) & 0xFFu) / 255.0;
+	col.blue = (tmp & 0xFFu) / 255.0;
 	col.alpha = 1.0;
 }
 
 int Util::gdkrgba_to_hex(GdkRGBA& color)
 {
-	return (((int)(color.red * 255)) & 0xff) << 16 |
-			(((int)(color.green * 255)) & 0xff) << 8 |
-			(((int)(color.blue * 255)) & 0xff);
+	return ((static_cast<unsigned>(color.red * 255)) & 0xffu) << 16u |
```

Comment:
Cast to unsigned and return as int makes the code only more complicated, and brings nothing. The same as above: Do you have a good reason?

Reply:
It removes false positive compiler warnings. Binary operators should not be used with unsigned values, because it is bug prone. That's why compilers and clang-tidy complain. Actually it would be better to return an unsigned, instead of an int. And i don't see a difference in readability between a cast to int or a cast to unsigned.